ConfigVersionControl/git_version_control.py

```diff
@@ -100,47 +102,70 @@ def branch_allowed(branch_name):
     def _unlock(self):
         """ Removes index.lock if it exists, and it's not being used
         """
-        try:
-            lock_file_path = os.path.join(self.repo.git_dir, "index.lock")
-            if os.path.exists(lock_file_path):
-                print_and_log("Found lock for version control repository, trying to remove: %s" % lock_file_path,
-                              "INFO")
-                os.remove(lock_file_path)
-                print_and_log("Lock removed from version control repository", "INFO")
-        except Exception as err:
-            raise UnlockVersionControlException("Unable to remove lock from version control repository: %s " %
-                                                err.message)
+        attempts = 0
+        while attempts < RETRY_MAX_ATTEMPTS:
+            try:
+                lock_file_path = os.path.join(self.repo.git_dir, "index.lock")
+                if os.path.exists(lock_file_path):
+                    print_and_log("Found lock for version control repository, trying to remove: %s" % lock_file_path,
+                                  "INFO")
+                    os.remove(lock_file_path)
+                    print_and_log("Lock removed from version control repository", "INFO")
+
+                return
+            except:
+                # Exception will be thrown below if the function doesn't return.
+                sleep(RETRY_INTERVAL)
+
+            attempts += 1
+
+        raise UnlockVersionControlException("Unable to remove lock from version control repository.")
 
     def add(self, path):
         """ Add a file to the repository
         Args:
             path (str): the file to add
         """
-        try:
-            if self._should_ignore(path):
-                return
-            self.repo.index.add([path])
-        except WindowsError as err:
-            # Most likely access denied, so try changing permissions then retry once
+        attempts = 0
+        while attempts < RETRY_MAX_ATTEMPTS:
             try:
-                self._set_permissions()
+                if self._should_ignore(path):
+                    return
                 self.repo.index.add([path])
+                return
+            except WindowsError as err:
+                # Most likely access denied, so try changing permissions then retry once
+                try:
+                    self._set_permissions()
+                    self.repo.index.add([path])
+                    return
+                except Exception as err:
+                    sleep(RETRY_INTERVAL)
             except Exception as err:
-                raise AddToVersionControlException(err.message)
-        except Exception as err:
-            raise AddToVersionControlException(err.message)
+                sleep(RETRY_INTERVAL)
+
+            attempts += 1
+
+            raise AddToVersionControlException("Couldn't add to version control")
```

Comment:
This line is indented too far, putting it inside the while loop, meaning the exception would be raised after the first failed attempt